minidump/src/minidump.rs

```diff
@@ -2592,7 +2610,12 @@ impl<'a> MinidumpThreadList<'a> {
 impl<'a> MinidumpStream<'a> for MinidumpSystemInfo {
     const STREAM_TYPE: MINIDUMP_STREAM_TYPE = MINIDUMP_STREAM_TYPE::SystemInfoStream;
 
-    fn read(bytes: &[u8], all: &[u8], endian: scroll::Endian) -> Result<MinidumpSystemInfo, Error> {
+    fn read(
+        bytes: &[u8],
+        all: &[u8],
+        endian: scroll::Endian,
+        _system_info: Option<&MinidumpSystemInfo>,
```

Comment:
Oh interesting! In my mind this was paradoxical and it would have to be factored out but I guess because you made it an Option it's fine.

Reply:
That said it would be aesthetically satisfying if this could short-circuit and return itself when it's Some.

Reply:
Yeah, that's an excellent idea.